### ISSUE TITLE
No packages are displayed within the console if we use the TAP 1.0.2 packages repository

### ISSUE THREAD

**Issue by cmoulliard** [vmware-tanzu/kubeapps#4524]
## Issue

No packages are displayed within the kubeapps console if we use the TAP 1.0.2 packages repository and latest kubeapps image published

![Screenshot 2022-03-29 at 12 29 29](https://user-images.githubusercontent.com/463790/160592197-906d64f5-e0fd-4a8d-83db-5ef00c65c42a.png)

Nevertheless, the different CRs reports correctly the information like the [packages.yaml](https://gist.github.com/cmoulliard/c46d96d6764db37f69ffdff0e5353f0c#file-packages-yaml) file extracted from the repository image
```
[snowdrop@h01-121 _temp]$ kc get pkgr -n tanzu
NAME                   AGE     DESCRIPTION
tanzu-tap-repository   9m39s   Reconcile succeeded

[snowdrop@h01-121 _temp]$ kc get pkg -n tanzu
NAME                                                        PACKAGEMETADATA NAME                                  VERSION         AGE
accelerator.apps.tanzu.vmware.com.1.0.0                     accelerator.apps.tanzu.vmware.com                     1.0.0           9m38s
accelerator.apps.tanzu.vmware.com.1.0.1                     accelerator.apps.tanzu.vmware.com                     1.0.1           9m38s
accelerator.apps.tanzu.vmware.com.1.0.2                     accelerator.apps.tanzu.vmware.com                     1.0.2           9m38s
api-portal.tanzu.vmware.com.1.0.8                           api-portal.tanzu.vmware.com                           1.0.8           9m37s
api-portal.tanzu.vmware.com.1.0.9                           api-portal.tanzu.vmware.com                           1.0.9           9m37s
build.appliveview.tanzu.vmware.com.1.0.1                    build.appliveview.tanzu.vmware.com                    1.0.1           9m39s
build.appliveview.tanzu.vmware.com.1.0.2                    build.appliveview.tanzu.vmware.com                    1.0.2           9m39s
build.appliveview.tanzu.vmware.com.1.0.2-build.2            build.appliveview.tanzu.vmware.com                    1.0.2-build.2   9m39s
buildservice.tanzu.vmware.com.1.4.2                         buildservice.tanzu.vmware.com                         1.4.2           9m37s
buildservice.tanzu.vmware.com.1.4.3                         buildservice.tanzu.vmware.com                         1.4.3           9m37s
cartographer.tanzu.vmware.com.0.1.0                         cartographer.tanzu.vmware.com                         0.1.0           9m36s
cartographer.tanzu.vmware.com.0.2.1                         cartographer.tanzu.vmware.com                         0.2.1           9m36s
cartographer.tanzu.vmware.com.0.2.2                         cartographer.tanzu.vmware.com                         0.2.2           9m36s
cert-manager.tanzu.vmware.com.1.5.3+tap.1                   cert-manager.tanzu.vmware.com                         1.5.3+tap.1     9m39s
cnrs.tanzu.vmware.com.1.1.0                                 cnrs.tanzu.vmware.com                                 1.1.0           9m37s
cnrs.tanzu.vmware.com.1.1.1                                 cnrs.tanzu.vmware.com                                 1.1.1           9m37s
contour.tanzu.vmware.com.1.18.2+tap.1                       contour.tanzu.vmware.com                              1.18.2+tap.1    9m39s
controller.conventions.apps.tanzu.vmware.com.0.5.0          controller.conventions.apps.tanzu.vmware.com          0.5.0           9m39s
controller.conventions.apps.tanzu.vmware.com.0.5.1          controller.conventions.apps.tanzu.vmware.com          0.5.1           9m39s
controller.source.apps.tanzu.vmware.com.0.2.0               controller.source.apps.tanzu.vmware.com               0.2.0           9m39s
controller.source.apps.tanzu.vmware.com.0.2.1               controller.source.apps.tanzu.vmware.com               0.2.1           9m39s
developer-conventions.tanzu.vmware.com.0.5.0                developer-conventions.tanzu.vmware.com                0.5.0           9m38s
developer-conventions.tanzu.vmware.com.0.5.0-build.1        developer-conventions.tanzu.vmware.com                0.5.0-build.1   9m38s
fluxcd.source.controller.tanzu.vmware.com.0.16.0            fluxcd.source.controller.tanzu.vmware.com             0.16.0          9m35s
fluxcd.source.controller.tanzu.vmware.com.0.16.1            fluxcd.source.controller.tanzu.vmware.com             0.16.1          9m35s
fluxcd.source.controller.tanzu.vmware.com.0.16.3            fluxcd.source.controller.tanzu.vmware.com             0.16.3          9m34s
grype.scanning.apps.tanzu.vmware.com.1.0.0                  grype.scanning.apps.tanzu.vmware.com                  1.0.0           9m38s
grype.scanning.apps.tanzu.vmware.com.1.0.1                  grype.scanning.apps.tanzu.vmware.com                  1.0.1           9m38s
image-policy-webhook.signing.apps.tanzu.vmware.com.1.0.0    image-policy-webhook.signing.apps.tanzu.vmware.com    1.0.0           9m37s
image-policy-webhook.signing.apps.tanzu.vmware.com.1.0.1    image-policy-webhook.signing.apps.tanzu.vmware.com    1.0.1           9m37s
image-policy-webhook.signing.apps.tanzu.vmware.com.1.0.2    image-policy-webhook.signing.apps.tanzu.vmware.com    1.0.2           9m37s
learningcenter.tanzu.vmware.com.0.1.0                       learningcenter.tanzu.vmware.com                       0.1.0           9m36s
learningcenter.tanzu.vmware.com.0.1.1                       learningcenter.tanzu.vmware.com                       0.1.1           9m36s
metadata-store.apps.tanzu.vmware.com.1.0.1                  metadata-store.apps.tanzu.vmware.com                  1.0.1           9m38s
metadata-store.apps.tanzu.vmware.com.1.0.2                  metadata-store.apps.tanzu.vmware.com                  1.0.2           9m38s
ootb-delivery-basic.tanzu.vmware.com.0.5.1                  ootb-delivery-basic.tanzu.vmware.com                  0.5.1           9m34s
ootb-delivery-basic.tanzu.vmware.com.0.6.1                  ootb-delivery-basic.tanzu.vmware.com                  0.6.1           9m34s
ootb-supply-chain-basic.tanzu.vmware.com.0.5.1              ootb-supply-chain-basic.tanzu.vmware.com              0.5.1           9m35s
ootb-supply-chain-basic.tanzu.vmware.com.0.6.1              ootb-supply-chain-basic.tanzu.vmware.com              0.6.1           9m35s
ootb-supply-chain-testing-scanning.tanzu.vmware.com.0.5.1   ootb-supply-chain-testing-scanning.tanzu.vmware.com   0.5.1           9m35s
ootb-supply-chain-testing-scanning.tanzu.vmware.com.0.6.1   ootb-supply-chain-testing-scanning.tanzu.vmware.com   0.6.1           9m35s
ootb-supply-chain-testing.tanzu.vmware.com.0.5.1            ootb-supply-chain-testing.tanzu.vmware.com            0.5.1           9m35s
ootb-supply-chain-testing.tanzu.vmware.com.0.6.1            ootb-supply-chain-testing.tanzu.vmware.com            0.6.1           9m35s
ootb-templates.tanzu.vmware.com.0.5.1                       ootb-templates.tanzu.vmware.com                       0.5.1           9m36s
ootb-templates.tanzu.vmware.com.0.6.1                       ootb-templates.tanzu.vmware.com                       0.6.1           9m35s
run.appliveview.tanzu.vmware.com.1.0.1                      run.appliveview.tanzu.vmware.com                      1.0.1           9m39s
run.appliveview.tanzu.vmware.com.1.0.2                      run.appliveview.tanzu.vmware.com                      1.0.2           9m39s
run.appliveview.tanzu.vmware.com.1.0.2-build.2              run.appliveview.tanzu.vmware.com                      1.0.2-build.2   9m39s
scanning.apps.tanzu.vmware.com.1.0.0                        scanning.apps.tanzu.vmware.com                        1.0.0           9m38s
scanning.apps.tanzu.vmware.com.1.0.1                        scanning.apps.tanzu.vmware.com                        1.0.1           9m38s
service-bindings.labs.vmware.com.0.6.0                      service-bindings.labs.vmware.com                      0.6.0           9m39s
service-bindings.labs.vmware.com.0.6.1                      service-bindings.labs.vmware.com                      0.6.1           9m39s
services-toolkit.tanzu.vmware.com.0.5.0                     services-toolkit.tanzu.vmware.com                     0.5.0           9m39s
services-toolkit.tanzu.vmware.com.0.5.1                     services-toolkit.tanzu.vmware.com                     0.5.1           9m39s
spring-boot-conventions.tanzu.vmware.com.0.3.0              spring-boot-conventions.tanzu.vmware.com              0.3.0           9m35s
tap-gui.tanzu.vmware.com.1.0.1                              tap-gui.tanzu.vmware.com                              1.0.1           9m35s
tap-gui.tanzu.vmware.com.1.0.2                              tap-gui.tanzu.vmware.com                              1.0.2           9m35s
tap-telemetry.tanzu.vmware.com.0.1.2                        tap-telemetry.tanzu.vmware.com                        0.1.2           9m34s
tap-telemetry.tanzu.vmware.com.0.1.3                        tap-telemetry.tanzu.vmware.com                        0.1.3           9m34s
tap-telemetry.tanzu.vmware.com.0.1.4                        tap-telemetry.tanzu.vmware.com                        0.1.4           9m39s
tap.tanzu.vmware.com.1.0.0                                  tap.tanzu.vmware.com                                  1.0.0           9m36s
tap.tanzu.vmware.com.1.0.1                                  tap.tanzu.vmware.com                                  1.0.1           9m36s
tap.tanzu.vmware.com.1.0.2                                  tap.tanzu.vmware.com                                  1.0.2           9m36s
tekton.tanzu.vmware.com.0.30.0                              tekton.tanzu.vmware.com                               0.30.0          9m34s
tekton.tanzu.vmware.com.0.30.1                              tekton.tanzu.vmware.com                               0.30.1          9m34s
workshops.learningcenter.tanzu.vmware.com.0.1.0             workshops.learningcenter.tanzu.vmware.com             0.1.0           9m36s
workshops.learningcenter.tanzu.vmware.com.0.1.1             workshops.learningcenter.tanzu.vmware.com             0.1.1           9m36s

[snowdrop@h01-121 _temp]$ kc get pkgm -n tanzu
NAME                                                  DISPLAY NAME                                                               CATEGORIES             SHORT DESCRIPTION                                    AGE
accelerator.apps.tanzu.vmware.com                     Application Accelerator for VMware Tanzu                                                          Used to create new projects and configurations.      10m0s
api-portal.tanzu.vmware.com                           API portal                                                                                        A unified user interface to enable search, disc...   9m59s
build.appliveview.tanzu.vmware.com                    Application Live View Conventions for VMware Tanzu                                                Application Live View convention server              10m1s
buildservice.tanzu.vmware.com                         Tanzu Build Service                                                                               Tanzu Build Service enables the building and au...   9m59s
cartographer.tanzu.vmware.com                         Cartographer                                                                                      Kubernetes native Supply Chain Choreographer.        9m58s
cnrs.tanzu.vmware.com                                 Cloud Native Runtimes                                                                             Cloud Native Runtimes is a serverless runtime b...   9m59s
controller.conventions.apps.tanzu.vmware.com          Convention Service for VMware Tanzu                                                               Convention Service enables app operators to con...   10m1s
controller.source.apps.tanzu.vmware.com               Tanzu Source Controller                                                                           Tanzu Source Controller enables workload create...   10m1s
developer-conventions.tanzu.vmware.com                Tanzu App Platform Developer Conventions                                                          Developer Conventions                                10m0s
fluxcd.source.controller.tanzu.vmware.com             Flux Source Controller                                                     policy                 The source-controller is a Kubernetes operator,...   9m57s
grype.scanning.apps.tanzu.vmware.com                  Grype for Supply Chain Security Tools - Scan                                                      Default scan templates using Anchore Grype           10m0s
image-policy-webhook.signing.apps.tanzu.vmware.com    Image Policy Webhook                                                       policy                 Image Policy Webhook enables defining of a poli...   9m59s
learningcenter.tanzu.vmware.com                       Learning Center for Tanzu Application Platform                             education,enablement   Guided technical workshops                           9m58s
metadata-store.apps.tanzu.vmware.com                  Supply Chain Security Tools - Store                                                               Post SBoMs and query for image, package, and vu...   10m0s
ootb-delivery-basic.tanzu.vmware.com                  Tanzu App Platform Out of The Box Delivery Basic                                                  Out of The Box Delivery Basic.                       9m56s
ootb-supply-chain-basic.tanzu.vmware.com              Tanzu App Platform Out of The Box Supply Chain Basic                                              Out of The Box Supply Chain Basic.                   9m57s
ootb-supply-chain-testing-scanning.tanzu.vmware.com   Tanzu App Platform Out of The Box Supply Chain with Testing and Scanning                          Out of The Box Supply Chain with Testing and Sc...   9m57s
ootb-supply-chain-testing.tanzu.vmware.com            Tanzu App Platform Out of The Box Supply Chain with Testing                                       Out of The Box Supply Chain with Testing.            9m57s
ootb-templates.tanzu.vmware.com                       Tanzu App Platform Out of The Box Templates                                                       Out of The Box Templates.                            9m58s
run.appliveview.tanzu.vmware.com                      Application Live View for VMware Tanzu                                                            App for monitoring and troubleshooting running ...   10m1s
scanning.apps.tanzu.vmware.com                        Supply Chain Security Tools - Scan                                                                Scan for vulnerabilities and enforce policies d...   10m0s
service-bindings.labs.vmware.com                      Service Bindings for Kubernetes                                                                   Service Bindings for Kubernetes implements the ...   10m1s
services-toolkit.tanzu.vmware.com                     Services Toolkit                                                           services               The Services Toolkit enables the management, li...   10m1s
spring-boot-conventions.tanzu.vmware.com              Tanzu Spring Boot Conventions Server                                                              Default Spring Boot convention server.               9m57s
tap-gui.tanzu.vmware.com                              Tanzu Application Platform GUI                                                                    web app graphical user interface for Tanzu Appl...   9m59s
tap-telemetry.tanzu.vmware.com                        Telemetry Collector for Tanzu Application Platform                                                Tanzu Application Plaform Telemetry                  9m56s
tap.tanzu.vmware.com                                  Tanzu Application Platform                                                                        Package to install a set of TAP components to g...   9m59s
tekton.tanzu.vmware.com                               Tekton Pipelines                                                                                  Tekton Pipelines is a framework for creating CI...   9m56s
workshops.learningcenter.tanzu.vmware.com             Workshop Building Tutorial                                                 education,enablement   Workshop Building Tutorial                           9m58s
```
## Log of kubeapps-internal-kubeappsapis pod

```
│ I0329 10:03:15.637349       1 root.go:37] kubeapps-apis has been configured with: core.ServeOptions{Port:50051, PluginDirs:[]string{"/plugins/kapp-controller", "/plugins/resources"}, ClustersConfigPath:"/config/clusters.con │
│ f", PluginConfigPath:"/config/kubeapps-apis/plugins.conf", PinnipedProxyURL:"http://kubeapps-internal-pinniped-proxy.kubeapps:3333", GlobalReposNamespace:"kubeapps", UnsafeLocalDevKubeconfig:false, QPS:50, Burst:100}        │
│ I0329 10:03:16.234578       1 plugins.go:151] Successfully registered plugin "/plugins/resources/resources-v1alpha1-plugin.so"                                                                                                  │
│ I0329 10:03:16.318956       1 server.go:158] Starting server on :50051                                                                                                                                                          │
│ I0329 10:04:27.034659       1 namespaces.go:24] +resources CheckNamespaceExists (cluster: "default", namespace="default")                                                                                                       │
│ I0329 10:04:27.058471       1 server.go:59] OK 23.91806ms /kubeappsapis.plugins.resources.v1alpha1.ResourcesService/CheckNamespaceExists                                                                                        │
│ I0329 10:04:27.200420       1 packages.go:169] +core GetInstalledPackageSummaries (cluster="default", namespace="default")                                                                                                      │
│ I0329 10:04:27.200590       1 server.go:59] OK 283.826µs /kubeappsapis.core.packages.v1alpha1.PackagesService/GetInstalledPackageSummaries                                                                                      │
│ I0329 10:04:29.257333       1 packages.go:59] +core GetAvailablePackageSummaries (cluster="default", namespace="default")                                                                                                       │
│ I0329 10:04:29.257391       1 server.go:59] OK 94.369µs /kubeappsapis.core.packages.v1alpha1.PackagesService/GetAvailablePackageSummaries                                                                                       │
│ I0329 10:04:39.704962       1 packages.go:169] +core GetInstalledPackageSummaries (cluster="default", namespace="tanzu")                                                                                                        │
│ I0329 10:04:39.705022       1 server.go:59] OK 102.521µs /kubeappsapis.core.packages.v1alpha1.PackagesService/GetInstalledPackageSummaries                                                                                      │
│ I0329 10:04:41.308421       1 packages.go:59] +core GetAvailablePackageSummaries (cluster="default", namespace="tanzu")                                                                                                         │
│ I0329 10:04:41.308549       1 server.go:59] OK 144.011µs /kubeappsapis.core.packages.v1alpha1.PackagesService/GetAvailablePackageSummaries
```
## Steps to reproduce the issue

```
VM_IP=0.0.0.0 # Change ME TO USE THE IP OF YOUR VM

cat <<EOF | kind create cluster --config=-
kind: Cluster
apiVersion: kind.x-k8s.io/v1alpha4
nodes:
- role: control-plane
  kubeadmConfigPatches:
  - |
    kind: InitConfiguration
    nodeRegistration:
      kubeletExtraArgs:
        node-labels: "ingress-ready=true"
  extraPortMappings:
  - containerPort: 80
    hostPort: 80
    protocol: TCP
  - containerPort: 443
    hostPort: 443
    protocol: TCP
EOF
kubectl apply -f https://github.com/vmware-tanzu/carvel-kapp-controller/releases/latest/download/release.yml
kubectl apply -f https://github.com/cert-manager/cert-manager/releases/download/v1.7.2/cert-manager.yaml
kubectl apply -f https://projectcontour.io/quickstart/contour.yaml
kubectl patch daemonsets -n projectcontour envoy -p '{"spec":{"template":{"spec":{"nodeSelector":{"ingress-ready":"true"},"tolerations":[{"key":"node-role.kubernetes.io/control-plane","operator":"Equal","effect":"NoSchedule"},{"key":"node-role.kubernetes.io/master","operator":"Equal","effect":"NoSchedule"}]}}}}'

kubectl create ns tanzu
cat <<EOF | kubectl apply -f -
apiVersion: packaging.carvel.dev/v1alpha1
kind: PackageRepository
metadata:
  name: tanzu-tap-repository
  namespace: tanzu
spec:
  fetch:
    imgpkgBundle:
      image: ghcr.io/halkyonio/tap-packages:1.0.2
EOF

helm repo add bitnami https://charts.bitnami.com/bitnami
cat <<EOF > kubeapps-values.yml
kubeappsapis:
  image:
    registry: docker.io
    repository: kubeapps/kubeapps-apis
    tag: latest
packaging:
  helm:
    enabled: false
  carvel:
    enabled: true 
featureFlags:
  operators: false
EOF

kubectl create ns kubeapps
helm install kubeapps -n kubeapps bitnami/kubeapps -f kubeapps-values.yml

cat <<EOF | kubectl apply -f -
apiVersion: projectcontour.io/v1
kind: HTTPProxy
metadata:
  name: kubeapps-grpc
  namespace: kubeapps
spec:
  virtualhost:
    fqdn: kubeapps.$VM_IP.nip.io
  routes:
    - conditions:
      - prefix: /apis/
      pathRewritePolicy:
        replacePrefix:
        - replacement: /
      services:
      - name: kubeapps-internal-kubeappsapis
        port: 8080
        protocol: h2c
    - services:
      - name: kubeapps
        port: 80
EOF
kubectl create --namespace default serviceaccount kubeapps-operator
kubectl create clusterrolebinding kubeapps-operator --clusterrole=cluster-admin --serviceaccount=default:kubeapps-operator
kubectl create clusterrolebinding kubeapps-operator-cluster-admin --clusterrole=cluster-admin --serviceaccount kubeapps:kubeapps-operator

kubectl get --namespace default secret $(kubectl get --namespace default serviceaccount kubeapps-operator -o jsonpath='{range .secrets[*]}{.name}{"\n"}{end}' | grep kubeapps-operator-token) -o jsonpath='{.data.token}' -o go-template='{{.data.token | base64decode}}' && echo

open http://kubeapps.$VM_IP.nip.io/#/c/default/ns/tanzu/catalog
```

**Comment by cmoulliard** [comment]
**REMARK**: The problem can also be reproduced using TCE repository !
```
kubectl delete pkgr/tanzu-tap-repository -n tanzu
kubectl create ns tanzu
cat <<EOF | kubectl apply -f -
apiVersion: packaging.carvel.dev/v1alpha1
kind: PackageRepository
metadata:
  name: tanzu-tap-repository
  namespace: tanzu
spec:
  fetch:
    imgpkgBundle:
      # image: ghcr.io/halkyonio/tap-packages:1.0.2
      image: projects.registry.vmware.com/tce/main
EOF
```

**Comment by cmoulliard** [comment]
**WARNING**: The problem (= empty catalog using TAP 1.0.2) do not exist if we use the latest released image - 2.4.3 of kubeapps and this config

```
cat <<EOF > kubeapps-values.yml
packaging:
  helm:
    enabled: false
  carvel:
    enabled: true 
featureFlags:
  operators: false
EOF
```

Pod image version -->
```
kubectl get pod/kubeapps-internal-kubeappsapis-75675c549d-9c5w5 -n kubeapps -o yaml
apiVersion: v1
kind: Pod
metadata:
...
  name: kubeapps-internal-kubeappsapis-75675c549d-9c5w5
  namespace: kubeapps
spec:
  containers:
  - args:
    - --plugin-dir
    - /plugins/kapp-controller
    - --plugin-dir
    - /plugins/resources
    - --clusters-config-path=/config/clusters.conf
    - --plugin-config-path=/config/kubeapps-apis/plugins.conf
    - --global-repos-namespace=kubeapps
    - --kube-api-qps=50.0
    - --kube-api-burst=100
    command:
    - /kubeapps-apis
    env:
    - name: GOGC
      value: "50"
    - name: PORT
      value: "50051"
    image: docker.io/bitnami/kubeapps-apis:2.4.3-debian-10-r44
```

![Screenshot 2022-03-29 at 12 54 31](https://user-images.githubusercontent.com/463790/160596448-74ad2f54-217b-49d8-9ac8-56c35844df78.png)

**Comment by absoludity** [comment]
When using your repo above for TAP - `ghcr.io/halkyonio/tap-packages:1.0.2`, I get the same error that I saw previously, because there is no package metadata for contour in that repo either (just like in the original tap one) as you can see in your paste in the description or by checking:

```
$ kubectl get pkg -n tanzu | grep contour
contour.tanzu.vmware.com.1.18.2+tap.1                       contour.tanzu.vmware.com                              1.18.2+tap.1    3m10s

$ kubectl get pkgm -n tanzu | grep contour
$ 
```
So, without updating Kubeapps to handle invalid repositories (ie. with packages without packagemetadata), I'm not sure what we expect here?

When I add the TCE repo as per your instructions above to the `tanzu` namespace, that's exactly what I see in Kubeapps:

![tanzu-catalog-in-tanzu-ns](https://user-images.githubusercontent.com/497518/160722111-016573e1-fc4d-44bf-bedb-795c1948db4a.png)

I'm going to update Kubeapps so it handles invalid packages (without corresponding packagemetadata) but otherwise, I can't reproduce the issue in plain Kind here. It could be because you're now running/testing with a combination of development (kubeapps-apis latest) and released (kubeapps dashboard) images. Not sure. But I can't reproduce the issue with a valid repo (such as TCE) with the devel images.

**Comment by absoludity** [comment]
Once I've done the fix to handle the invalid repo, I'll get you to recheck, if that's OK :)

**Comment by absoludity** [comment]
So #4527 enables me to see the packages from the TAP repository in the tanzu namespace, as shown on that PR. I'm not 100% that it's the same cause for what you see, but given you're running with a mix of released and devel images, it's hard to know for sure. Feel free to try building the image from this branch (`IMAGE_TAG=dev1 make kubeapps/kubeapps-apis`), load it onto your kind node and update your deployment to try it out, but otherwise it can wait for the next release.

**Comment by cmoulliard** [comment]
> So, without updating Kubeapps to handle invalid repositories (ie. with packages without packagemetadata), I'm not sure what we expect here?

As there is a relation 1-1 between the PackageMetadata and a Package, then we should perhaps display an icon or an error for that specific package within the screen catalog when its packageMetadata is missing. Of course, all the packages where a PackageMetadata is there should be displayed. 

WDYT ? @absoludity

**Comment by cmoulliard** [comment]
> It could be because you're now running/testing with a combination of development (kubeapps-apis latest) and released (kubeapps dashboard) images

I use exactly the instructions as reported top of the ticket and where the kubeapps/kubeapps-apis:latest (= the one published yesterday on dockerhub)

**Comment by absoludity** [comment]
Yep, that's what I meant: a mix of development and released images (you have kubeapps-apis with the latest developments image to test it, but all other services, such as the dashboard, with the images from the previous release). It is just a difference between our environments which I haven't checked yet. Let me know if you get to try the change in the related PR.

**Comment by cmoulliard** [comment]
I did a new test using PR4527 and the screen catalog is still empty as reported top of this ticket. Here are the steps I followed to build kubeapps and configure it

```
git clone https://github.com/kubeapps/kubeapps.git && cd kubeapps/
git fetch origin pull/4527/head:pr4527 && git checkout pr4527
IMAGE_TAG=dev1 make kubeapps/kubeapps-apis
docker tag kubeapps/kubeapps-apis:dev1 ghcr.io/halkyonio/kubeapps-apis:dev1
docker push ghcr.io/halkyonio/kubeapps-apis:dev1
```
Chart cfg
```
kubeappsapis:
  image:
    registry: ghcr.io
    repository: halkyonio/kubeapps-apis
    tag: dev1
packaging:
  helm:
    enabled: false
  carvel:
    enabled: true
featureFlags:
  operators: false
```
It has been redeployed here: http://kubeapps.65.108.148.216.nip.io/#/c/default/ns/tanzu/catalog.

**Comment by absoludity** [comment]
Sorry, hadn't meant to close this until you could confirm. Yep, it fixes my local Kind env running all the development images. I need to dig further to see what the difference is between our environments that could cause the difference (starting with the mixture of images mentioned above).

**Comment by absoludity** [comment]
I got a bit of time tonight to debug the local environment. It was as I suspected an issue due to the mixing of the released chart with the devel image. The issue was that the `kapp-controller` plugin was not actually being loaded nor running. You can actually see this in the logs of the pod which you pasted in the description, but I didn't notice it at the time. You'll see there that only the `resources` plugin is loaded. This is due to a difference in the plugin name in the upstream chart and image (which we were already aware of and fixing for the next release). On the development image of the `kubeapps-apis` service, the plugin has the suffix `-packages`. Anyway, to fix it I just explicitly set the list of plugins using the correct name for the dev image:

```
cat <<EOF > kubeapps-values.yml
kubeappsapis:
  image:
    registry: docker.io
    repository: kubeapps/kubeapps-apis
    tag: latest
  enabledPlugins:
    - resources
    - kapp-controller-packages
featureFlags:
  operators: false
EOF

helm delete kubeapps -n kubeapps && helm install kubeapps -n kubeapps bitnami/kubeapps -f kubeapps-values.yml
```

With that change, you'll then see the error that I have fixed in #4527 , so when you subsequently update to use your image based on that PR, it works:

![tap-catalog](https://user-images.githubusercontent.com/497518/160812271-0ec7ff31-8dd8-43e1-9b77-f92a8f96abe6.png)

If you verify that you see the catalog correctly with that config change and the new image, please feel free to close this issue.

Note: these plugins supporting flux and carvel are quite new, so you may find other issues, but please open them as separate issues. Thanks for all your input Charles!

**Comment by absoludity** [comment]
I haven't, but I haven't tried with 22 installed packages either. I did some work to optimise the available packages view earlier as it was very slow, and the same optimisation can be done for the installed packages, we just haven't prioritised it. Can you create a separate issue please? I'll close this one now, since the combination of the fix in #4527 and the config change solve it. Thanks!
